### DISCUSSION TITLE
Examples including Lambda powertools and xray

### DISCUSSION THREAD

**Discussion by Denn0** [aws-samples/serverless-test-samples#301] [Ideas]
Hi, 

I'm moving from Python to TypeScript as I moved to another team recently. Kind of struggling with some of the auxiliary tool in lambda, such as [X-Ray](https://docs.aws.amazon.com/lambda/latest/dg/typescript-tracing.html#typescript-tracing-cdk) and [Powertools](https://docs.powertools.aws.dev/lambda/typescript/latest/). In my experience Python is a bit more lenient, where typescript is a bit more strict and then we have the transpilation step to cope with. 

AWS Docs prescribes to use [Middy middleware module](https://middy.js.org/) to [inject lambda context into the logger](https://docs.powertools.aws.dev/lambda/typescript/latest/core/logger/#log-incoming-event). That works fine, but now I run into unittest  issues due to using middy. 

Could you guys perhaps provide some Typescript unittest examples that includes testing a lambda that uses Powertools (with middy) and X-Ray functionality? 

Would be much appreciated! 

Dennis

**Comment by brnkrygs** [comment]
Hi @Denn0,

Thanks for the feedback! We'll get a task on our backlog to provide an example like this.

Are there specific services (besides AWS Lambda of course!) that you'd like to see included? Is there a specific scenario you're thinking about (shopping cart, admin Create Read Update Delete, IoT event processing, etc)?

We can do something generic of course, but we like making examples that are more "real-world" when possible.

**Reply by Denn0** [reply]
Thanks for your quick response! 
Looking at the existing examples, why not include an event bridge example? That one is not yet there atm. 
IoT is interesting as well ofc. No real preference there.
